pkg/logs/pipeline/provider.go

```diff
@@ -131,28 +133,43 @@ func (p *provider) reconfigureSDS(config []byte, orderType sds.ReconfigureOrderT
 	// reports if at least one error occurred
 
 	var rerr error
+	allScannersActive := true
 	for _, response := range responses {
-		err := <-response
-		if err != nil {
-			rerr = multierror.Append(rerr, err)
+		resp := <-response
+
+		if !resp.IsActive {
+			allScannersActive = false
+		}
+
+		if resp.Err != nil {
+			rerr = multierror.Append(rerr, resp.Err)
 		}
+
 		close(response)
 	}
 
-	return rerr
+	return allScannersActive, rerr
 }
 
 // ReconfigureSDSStandardRules stores the SDS standard rules for the given provider.
-func (p *provider) ReconfigureSDSStandardRules(standardRules []byte) error {
+func (p *provider) ReconfigureSDSStandardRules(standardRules []byte) (bool, error) {
 	return p.reconfigureSDS(standardRules, sds.StandardRules)
 }
 
 // ReconfigureSDSAgentConfig reconfigures the pipeline with the given
 // configuration received through Remote Configuration.
-func (p *provider) ReconfigureSDSAgentConfig(config []byte) error {
+// Return true if all SDS scanners are active after applying this configuration.
+func (p *provider) ReconfigureSDSAgentConfig(config []byte) (bool, error) {
 	return p.reconfigureSDS(config, sds.AgentConfig)
 }
 
+// StopSDSProcessing reconfigures the pipeline removing the SDS scanning
+// from the processing steps.
+func (p *provider) StopSDSProcessing() error {
+	_, err := p.reconfigureSDS([]byte("{}"), sds.StopProcessing)
```

Comment:
This gives an impression that the config passed is meaningful. Would it make sense to pass nil here instead?

```suggestion
	_, err := p.reconfigureSDS([]byte("{}"), sds.StopProcessing)
```

(A safer approach would be to send different message structs that don't contain the fields that won't be used).

Reply:
Considered it as well, I'll do it since it seems to be a shared opinion 👍 I'll fix this